Source/cmake/OptionsHaiku.cmake

```diff
@@ -189,6 +189,7 @@ SET_AND_EXPOSE_TO_BUILD(ENABLE_GRAPHICS_CONTEXT_GL OFF)
 SET_AND_EXPOSE_TO_BUILD(USE_EGL FALSE)
 SET_AND_EXPOSE_TO_BUILD(USE_TILED_BACKING_STORE OFF)
 SET_AND_EXPOSE_TO_BUILD(USE_COORDINATED_GRAPHICS OFF)
+SET_AND_EXPOSE_TO_BUILD(HAVE_OS_DARK_MODE_SUPPORT 1)
```

Comment:
It looks like this should instead be set in Source/WTF/wtf/PlatformHave.h instead (just add PLATFORM(HAIKU) to the tests there), and additionally set it as a CSS_VALUE_PLATFORM_DEFINES in Sources/WebCore/PlatformHaiku.cmake (as it is done for GTK and WPE).

The HAVE_ macros are in PlatformHave and not configurable, while the USE_ macros can be enabled/disabled at build time by passing arguments to cmake.

Reply:
If I understand this correctly, OptionsHaiku is for things that can be configured from cmake side (for example using -D options on the command line, or one of the cmake user interfaces like ccmake). PlatformHaiku is for things that are always enabled or enabled by already existing options.

But actually, you're right, GTK also has it in its Options file, so we probably need all of these. I guess no one is really sure how this all works and now the same thing has to be defined in three different files :'(